bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -247,78 +250,7 @@ public void configurationUpdated(Thing thing) {
-                            configDescriptionRegistry);
-                    ((ThingImpl) thing).setChannels(channels);
-
-                    // Set the given configuration
-                    ThingFactoryHelper.applyDefaultConfiguration(configuration, thingType, configDescriptionRegistry);
-                    ((ThingImpl) thing).setConfiguration(configuration);
-
-                    // Change the ThingType
-                    ((ThingImpl) thing).setThingTypeUID(thingTypeUID);
-
-                    // Register the new Handler - ThingManager.updateThing() is going to take care of that
-                    thingRegistry.update(thing);
-
-                    logger.debug("Changed ThingType of Thing {} to {}. New ThingHandler is {}.",
-                            thing.getUID().toString(), thing.getThingTypeUID(), thing.getHandler().toString());
-                }
-
-                private void waitUntilHandlerUnregistered(final Thing thing, int timeout) {
-                    for (int i = 0; i < timeout / 100; i++) {
-                        if (thing.getHandler() == null && thingHandlers.get(thing.getUID()) == null) {
-                            return;
-                        }
-                        try {
-                            Thread.sleep(100);
-                            logger.debug(
-                                    "Waiting for handler deregistration to complete for thing {}. Took already {}ms.",
-                                    thing.getUID().getAsString(), i * 100);
-                        } catch (InterruptedException e) {
-                            return;
-                        }
-                    }
-                    throw new RuntimeException(MessageFormat.format(
-                            "Thing type migration failed for {0}. The handler deregistration did not complete within {1}ms.",
-                            thing.getUID().getAsString(), timeout));
-                }
-
-                private void waitForRunningHandlerRegistrations(ThingUID thingUID) {
-                    for (int i = 0; i < 10 * 10; i++) {
-                        if (!registerHandlerLock.contains(thingUID)) {
-                            return;
-                        }
-                        try {
-                            // Wait a little to give running handler registrations a chance to complete...
-                            Thread.sleep(100);
-                        } catch (InterruptedException e) {
-                            return;
-                        }
-                    }
-                    throw new RuntimeException(MessageFormat.format(
-                            "Thing type migration failed for {0}. Could not obtain lock for hander registration.",
-                            thingUID.getAsString()));
-                }
-            }, 0, TimeUnit.MILLISECONDS);
+            ThingManager.this.changeThingType(thing, thingTypeUID, configuration);
```

Comment:
if we decide to rename changeThingType to migrateThingType for ThingTypeMigrator then we should think about to rename the callback operation as well